src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -0,0 +1,489 @@
+                EigenPodManager.initialize.selector,
+                type(uint256).max, // maxPods
+                address(beaconChainOracle),
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0 // initialPausedStatus
+            )
+        );
+        // Delayed Withdrawal Router
+        uint256 withdrawalDelayBlocks = 7 days / 12 seconds;
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))),
+            address(delayedWithdrawalRouterImplementation),
+            abi.encodeWithSelector(
+                DelayedWithdrawalRouter.initialize.selector,
+                eigenLayerReputedMultisig, // initialOwner
+                pauserRegistry,
+                0, // initialPausedStatus
+                withdrawalDelayBlocks
+            )
+        );
+
+        // Create base strategy implementation and deploy a few strategies
+        baseStrategyImplementation = new StrategyBase(strategyManager);
+
+        _newStrategyAndToken("Strategy1Token", "str1", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy2Token", "str2", 10e50, address(this)); // initialSupply, owner
+        _newStrategyAndToken("Strategy3Token", "str3", 10e50, address(this)); // initialSupply, owner
+
+        ethStrats.push(BEACONCHAIN_ETH_STRAT);
+        mixedStrats.push(BEACONCHAIN_ETH_STRAT);
+
+        global = new Global();
+    }
+
+    /// @dev Deploy a strategy and its underlying token, push to global lists of tokens/strategies, and whitelist
+    /// strategy in strategyManager
+    function _newStrategyAndToken(string memory tokenName, string memory tokenSymbol, uint initialSupply, address owner) internal {
+        IERC20 underlyingToken = new ERC20PresetFixedSupply(tokenName, tokenSymbol, initialSupply, owner); 
+        StrategyBase strategy = StrategyBase(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(baseStrategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, underlyingToken, pauserRegistry)
+                )
+            )
+        );
+
+        // Push to list of tokens/strategies
+        _strategies.push(strategy);
+
+        // Whitelist strategy
+        IStrategy[] memory strategies = new IStrategy[](1);
+        strategies[0] = strategy;
+        strategyManager.addStrategiesToDepositWhitelist(strategies);
+
+        // Add to lstStrats and mixedStrats
+        lstStrats.push(strategy);
+        mixedStrats.push(strategy);
+    } 
+
+    function _configRand(
+        uint16 _randomSeed, 
+        uint _assetTypes,
+        uint _signedTypes
+    ) internal {
+        // Using uint16 for the seed type so that if a test fails, it's easier
+        // to manually use the seed to replay the same test.
```

Comment:
not sure I'm understanding this comment. You should be able to run a fuzzed test with a particular input using syntax like:
```
forge script src/test/unit/DelegationUnit.t.sol:DelegationUnitTests --sig "testDecreaseDelegatedShares(address,address[],uint128,bool)" 0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f "[0x86294a1f20Afb94b41fbF27bd26930729C967dC2, 0x00000000000000000000000000000000000007D8, 0x02f460f68b8d2Bf4D66903FA7eb9F6a172b0FA20, 0x000000000000000000000000000000000000133C, 0x0000000000000000000000000000000000002616, 0x0000000000000000000000000000000000002925, 0x0000000000000000000000000000000000004baB, 0x27EA5aD96722c4D148c38Fb89BeE32f64F0e6065, 0x6D69979669F28Ac041dFBa0B2a781fe6f366E474, 0x00000000000000604A3ac11ce6251dCA61A05869]" 8843 true
```

I would think a uint256 seed would be best, to capture more scenarios / so it could be chopped up into more slices

Reply:
i think what i'm noticing is that when a test fails, it gives you the input it failed with - like the input was "0" or "1" or whatever. if the typing is a low-size uint, those values are readable and easy to quickly spin up a test with the same inputs. i was worried that a high-size input would require a lot more parsing and care?

we could definitely increase the size of `uint16`, but it's only running 256 scenarios regardless

Reply:
[c99e847](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/c99e847709852d7246c73b7d72d44bba368b760e)

increased to uint24